functions/index.ts

```diff
@@ -1,87 +1,83 @@
 const AWS = require('aws-sdk');
-import { Game, GetGame, ModifyGame, DeleteGame, IHttpResponse, IDynamoObject, ListGames } from "./game"
+import { Game } from "./game"
+import { getGame, listGames, IHttpResponse, IJSONPayload, deleteGame, modifyGame } from "./gameManager";
 
 exports.handler = async (event: any, context: any, callback: any) => {
+  const userID = event.requestContext.authorizer.claims['cognito:username'];
   switch (event.path) {
     case ("/getGame"):
-      let getGameData = DeserializeGameData({gameName: event.queryStringParameters["gameName"]});
-      callback(null, await GetGameHttpResponse(getGameData));
+      let gameName = { gameName: event.queryStringParameters["gameName"] };
+      let getGameData = deserializeGameData(userID, gameName)
+      callback(null, await getGameHttpResponse(getGameData));
       break;
-    case ("/listGames"):
-      callback(null, await ListGamesHttpResponse());
+    case("/listGames"):
+      let listGameData = userID;
+      callback(null, await listGamesHttpResponse(listGameData));
       break;
     case("/createGame"):
-      let createGameData = DeserializeGameData(JSON.parse(event.body));
-      callback(null, await CreateGameHttpResponse(createGameData));
+      let createGameData = deserializeGameData(userID, JSON.parse(event.body));
+      callback(null, await createGameHttpResponse(createGameData));
       break;
     case("/modifyGame"):
-      let modifyGameData = DeserializeGameData(JSON.parse(event.body));
-      callback(null, await ModifyGameHttpResponse(modifyGameData));
+      let modifyGameData = deserializeGameData(userID, JSON.parse(event.body));
+      callback(null, await modifyGameHttpResponse(modifyGameData));
       break;
     case ("/deleteGame"):
-      let deleteGameData = DeserializeGameData(JSON.parse(event.body));
-      callback(null, await DeleteGameHttpResponse(deleteGameData));
+      let deleteGameData = deserializeGameData(userID, JSON.parse(event.body));
+      callback(null, await deleteGameHttpResponse(deleteGameData));
       break;
     default:
-      callback(null, HttpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
+      callback(null, httpResponse({statusCode: 400, body: JSON.stringify("Invalid operation.")}));
       break;
     }
 }  
 
-export async function GetGameHttpResponse(game: Game) {
-  let response = await GetGame(game);
-  if (response.code) {
-    return ParseDynamoError(response.code);
-  } else if (response.Item) {
-    return HttpResponse({statusCode: 200, body: JSON.stringify(response)});
-  } else {
-    return HttpResponse({statusCode: 404, body: "Unable to get game."});
+export async function getGameHttpResponse(game: Game) {
+  try {
+    let response = await getGame(game);
+    return httpResponse({statusCode: 200, body: JSON.stringify(response)});
+  } catch (err: any) {
+    return httpResponse({statusCode: err.statusCode, body: err.message});
```

Comment:
I would filter here to see if its an instance of GameError, if so, return the code and message, otherwise just return a 500 and a generic message like "Oops something bad happened on our side".